src/D2L.CodeStyle.Analyzers/Async/Generator/SyncGenerator.cs

```diff
@@ -0,0 +1,182 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+[Generator]
+internal sealed partial class SyncGenerator : IIncrementalGenerator {
+	public void Initialize( IncrementalGeneratorInitializationContext context ) {
+		// Collect all the methods we want to generate sync versions for
+		// individually (for better incremental builds)
+		IncrementalValuesProvider<(MethodDeclarationSyntax, Compilation)> methodsToGenerate =
+			context.SyntaxProvider
+			   .CreateSyntaxProvider( IsInterestingLookingMethod, ExtractSyntax )

```

Comment:
For my reading benefit, not knowing the APIs uber well

```suggestion
			   .CreateSyntaxProvider( predicate: IsInterestingLookingMethod, transform: ExtractSyntax )
```